bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/api/DoorbirdAPI.java

```diff
@@ -0,0 +1,255 @@
+        this.httpClient = httpClient;
+    }
+
+    public @Nullable DoorbirdInfo getDoorbirdInfo() {
+        DoorbirdInfo doorbirdInfo = null;
+        try {
+            String infoResponse = executeGetRequest("/bha-api/info.cgi");
+            logger.debug("Doorbird returned json response: {}", infoResponse);
+            doorbirdInfo = new DoorbirdInfo(infoResponse);
+        } catch (IOException e) {
+            logger.info("Unable to communicate with Doorbird: {}", e.getMessage());
+        } catch (JsonSyntaxException e) {
+            logger.info("Unable to parse Doorbird response: {}", e.getMessage());
+        } catch (DoorbirdUnauthorizedException e) {
+            logAuthorizationError("getDoorbirdName");
+        }
+        return doorbirdInfo;
+    }
+
+    public @Nullable SipStatus getSipStatus() {
+        SipStatus sipStatus = null;
+        try {
+            String statusResponse = executeGetRequest("/bha-api/sip.cgi&action=status");
+            logger.debug("Doorbird returned json response: {}", statusResponse);
+            sipStatus = new SipStatus(statusResponse);
+        } catch (IOException e) {
+            logger.info("Unable to communicate with Doorbird: {}", e.getMessage());
+        } catch (JsonSyntaxException e) {
+            logger.info("Unable to parse Doorbird response: {}", e.getMessage());
+        } catch (DoorbirdUnauthorizedException e) {
+            logAuthorizationError("getSipStatus");
+        }
+        return sipStatus;
+    }
+
+    public void lightOn() {
+        try {
+            String response = executeGetRequest("/bha-api/light-on.cgi");
+            logger.debug("Response={}", response);
+        } catch (IOException e) {
+            logger.debug("IOException turning on light: {}", e.getMessage());
+        } catch (DoorbirdUnauthorizedException e) {
+            logAuthorizationError("lightOn");
+        }
+    }
+
+    public void restart() {
+        try {
+            String response = executeGetRequest("/bha-api/restart.cgi");
+            logger.debug("Response={}", response);
+        } catch (IOException e) {
+            logger.debug("IOException restarting device: {}", e.getMessage());
+        } catch (DoorbirdUnauthorizedException e) {
+            logAuthorizationError("restart");
+        }
+    }
+
+    public void sipHangup() {
+        try {
+            String response = executeGetRequest("/bha-api/sip.cgi?action=hangup");
+            logger.debug("Response={}", response);
+        } catch (IOException e) {
+            logger.debug("IOException hanging up SIP call: {}", e.getMessage());
+        } catch (DoorbirdUnauthorizedException e) {
+            logAuthorizationError("sipHangup");
+        }
+    }
+
+    public @Nullable DoorbirdImage downloadCurrentImage() {
+        return downloadImage("/bha-api/image.cgi");
+    }
+
+    public @Nullable DoorbirdImage downloadDoorbellHistoryImage(String imageNumber) {
```

Comment:
If the parameter is always a number, why not just make it type `int` instead of type `String`?

Reply:
Since it needs to be converted to a String when building the URL, I thought it best to leave it as a String. I also wasn't sure if it always would always be a number in future Doorbirds.